karapace/master_coordinator.py

```diff
@@ -212,6 +222,16 @@ def init_schema_coordinator(self) -> None:
         )
         self.schema_coordinator_ready.set()
 
+    def get_coordinator_status(self) -> SchemaCoordinatorStatus:
+        generation = self.sc.generation() if self.sc is not None else None
+        return SchemaCoordinatorStatus(
+            is_primary=self.sc.are_we_master if self.sc is not None else None,
+            is_primary_eligible=cast(bool, self.config["master_eligibility"]),
+            primary_url=self.sc.master_url if self.sc is not None else None,
+            is_running=self.schema_coordinator_ready.is_set(),
```

Comment:
The ready flag been set is known but not if the thread is actually still running, hanging or failing. The more correct would be `self.running`, but it does not also tell if the loop is performing.

Reply:
Good point. Looking more, I'm thinking if `Thread.is_alive()` would serve best way here...